File: misc/aooswd/ex/facade/facade.java
public class Compiler
{
	public Compiler ();
	public void Compile (InputStream, ByteCodeStream)
	{
		Scanner scanner = new Scanner (InputStream);
		ProgramNodeBuilder builder = new ProgramNodeBuilder ();
		Parser parser = new parser ();
		parser.Parse (scanner, builder);
		RISCCodeGenerator generator = new generator (output);
		ProgramNode parseTree = builder. GetRootNode ();
		parseTree. Traverse (generator);
	}
}

could change to take components as arguments if components could vary

File: misc/aooswd/ex/facade/facade.cpp
class Compiler
{
	Compiler ();
	virtual void Compile (istream&, ByteCodeStream&);
};

void Compiler::Compile (istream& input, ByteCodeStream& output)
{
	Scanner scanner (input);
	ProgramNodeBuilder builder;
	Parser parser;
	parser. Parse (scanner, builder);
	RISCCodeGenerator generator (output);
	ProgramNode* parseTree = builder. GetRootNode ();
	parseTree -> Traverse (generator);
}

could change to take components as arguments if components could vary